lib/puppet.rb

```diff
@@ -36,6 +38,27 @@ module Puppet
   require 'puppet/environments'
 
   class << self
+    # TODO: (PUP-6958) Handle the rubygems case
+    local_locale_path = File.absolute_path('../locales', File.dirname(__FILE__))
+    posix_system_locale_path = File.absolute_path('../../../share/locale', File.dirname(__FILE__))
+    win32_system_locale_path = File.absolute_path('../../../../../puppet/share/locale', File.dirname(__FILE__))
```

Comment:
Not a merge blocker or anything but it would be nice (IMO) to have comments here with examples of these expanded out in the default case